src/app/launcher/create-app/mission-runtime-createapp-step/mission-runtime-createapp-step.component.ts

```diff
@@ -94,398 +139,153 @@ export class MissionRuntimeCreateappStepComponent extends LauncherStep implement
-              if (version !== undefined && version.id === 'community') {
-                return true;
-              }
-            }
-          }
-        }
-      }
-    } else {
-      return false;
-    }
+  resetSelections(): void {
+    this.clearMission();
+    this.clearRuntime();
+    this.updateSelection();
+    this.initCompleted();
   }
 
-  /**
-   * Returns true if mission choice should be disabled
-   *
-   * @param {Mission} mission The mission choice to test
-   * @returns {boolean} True if mission choice should be disabled
-   */
-  isMissionDisabled(mission: Mission): boolean {
-    if (this.launcherComponent.flow === 'osio') {
-      return this.isMissionDisabledOSIO(mission);
-    }
-    let runtime = this.launcherComponent.summary.runtime; // selected runtime
-    if (runtime === undefined) {
-      return false; // Nothing should be disabled initially
-    }
-
-    let result = true;
-    for (let i = 0; i < mission.runtimes.length; i++) {
-      if (mission.runtimes[i] === runtime.id) {
-        result = false;
-        break;
-      }
-    }
+  selectMission(mission: ViewMission): void {
+    this.missionId = mission.id;
+    this.launcherComponent.summary.mission = {
+      id: mission.id,
+      name: mission.name,
+      description: mission.description
+    };
+    this.updateSelection();
+  }
 
-    // Ensure selected version is complatible with mission
-    if (!result) {
-      result = true;
-      if (runtime.version === undefined) {
-        result = false;
-      } else {
-        for (let i = 0; i < runtime.missions.length; i++) {
-          if (mission.id === runtime.missions[i].id) {
-            for (let k = 0; k < runtime.missions[i].versions.length; k++) {
-              let version = runtime.missions[i].versions[k];
-              if (version !== undefined && version.id === runtime.version.id) {
-                result = false;
-                break;
-              }
-            }
-            break;
-          }
-        }
-      }
+  selectRuntime(runtime: ViewRuntime, version?: BoosterVersion): void {
+    this.runtimeId = runtime.id;
+    const newVersion =  version ? version : runtime.selectedVersion;
+    this.versionId = newVersion.id;
+    this.launcherComponent.summary.runtime = {
+      id: runtime.id,
+      name: runtime.name,
+      description: runtime.description,
+      icon: runtime.icon,
+      version: newVersion
+    };
+    this.updateSelection();
+
+    // FIXME: use a booster change event listener to do this
+    // set maven artifact
+    if (this.stepCompleted) {
+      let artifactTS: Date = new Date();
+      let runtime = this.launcherComponent.summary.runtime.id.replace(/[.\-_]/g, '');
+      let mission = this.launcherComponent.summary.mission.id.replace(/[.\-_]/g, '');
+      this.launcherComponent.summary.dependencyCheck.mavenArtifact = 'booster' + '-' + mission + '-' + runtime
+        + '-' + artifactTS.getTime();
```

Comment:
this timestamp should only be added for osio

Reply:
Yes I will do that using the broadcaster :)